### ISSUE TITLE
Exporting instance of the container gives undefined 

### ISSUE THREAD

**Issue by sharifismailk** [jeffijoe/awilix#318]
`import { Server } from 'socket.io';

import { configure } from './configure';

const io = new Server();
const container = configure(io);

export { container };
`
Here is the User router file where I tried importing it.

`import { Router } from 'express';
import { container } from '../index';

const router = Router();

const userService = container.resolve('userService');

router.get('/users', async (req, res) => {
  const users = await userService.getAll();
  res.json(users);
});

export { router };
`

**Comment by jeffijoe** [comment]
There’s not enough info for me to have any chance of helping you. However just from the title alone it sounds like its not an Awilix issue but an issue with modules like circular references.

**Comment by sharifismailk** [comment]
Yes this is an issue with my code not the library. I just wanted help because it's the container instance being undefined

**Comment by sharifismailk** [comment]
Let me upload the code snippets

**Comment by sharifismailk** [comment]
I finally got around it, thanks 🙏

**Comment by sharifismailk** [comment]
Actually, I don't know what happening with my code. Look at this, it's like I'm not allowed to export this container this way.

**Comment by sharifismailk** [comment]
But this listings router file can't resolve the imported container, for some reason container instance here is undefined, and gives the error; 
TypeError: Cannot read properties of undefined (reading 'build')

![listings-file](https://user-images.githubusercontent.com/76264952/229717213-87d95c59-80ef-4c18-aab9-2265dbcd190e.png)

**Comment by sharifismailk** [comment]
@jeffijoe what could be the problem

**Comment by jeffijoe** [comment]
Closing because this is still not an Awilix issue.

Again, I am almost certain you have a circular reference in your imports, like if your `SocketController` imports `container` and `container` imports `SocketController`, since `SocketController` runs while `container` is being set up, `container` won't be defined. You need to rearchitect your code. 

[I have a boilerplate available](https://github.com/jeffijoe/koa-es7-boilerplate), it's outdated in terms of dependencies and build setup, but the overall structure is still good, and my general recommendation to use Koa over Express still stands.

**Comment by sharifismailk** [comment]
Thanks @jeffijoe
The issue was indeed circular dependencies, I eliminated those with a little bit of refactoring, en boom. Thanks again
